ttt.c

```diff
@@ -10,16 +10,35 @@
 
 #define MAX_SEARCH_DEPTH 6
 #define BOARD_SIZE 4
-#define N_GRIDS (BOARD_SIZE) * (BOARD_SIZE)
-#define GET_INDEX(i, j) (i) * (BOARD_SIZE) + (j)
+#define GOAL 3
+#define ALLOW_EXCEED 1
+#define N_GRIDS ((BOARD_SIZE) * (BOARD_SIZE))
+#define GET_INDEX(i, j) ((i) * (BOARD_SIZE) + (j))
 #define GET_COL(x) ((x) % (BOARD_SIZE))
 #define GET_ROW(x) ((x) / (BOARD_SIZE))
-
-_Static_assert(BOARD_SIZE < 26, "Board size must be less than 26");
-
-enum { ROW, COL, PRIMARY, SECONDARY };
-static int no_step[2] = {0, 0};
-static const int step_forward[2] = {1, 2}, step_backward[2] = {-1, -2};
+#define LOOKUP(table, i, j, else_value)                             \
+    ((i) < 0 || (j) < 0 || (i) > (BOARD_SIZE) || (j) > (BOARD_SIZE) \
+         ? (else_value)                                             \
+         : (table)[GET_INDEX(i, j)])
+
+_Static_assert((BOARD_SIZE) <= 26, "Board size must not be greater than 26");
```

Comment:
It is not necessary to have extra parentheses. That is, write as `_Static_assert(BOARD_SIZE <= 26, ...)`.